site/docs/guide/files.md

```diff
@@ -6,41 +6,66 @@ This involves handling the files that are attached to the messages.
 ## How Files Work for Telegram Bots
 
 > This section explains how files work for Telegram bots.
-> If you want to know how you can work with files in grammY scroll down for [downloading](#receiving-files) and [uploading](#sending-files) files.
+> If you want to know how you can work with files in grammY, scroll down for [downloading](#receiving-files) and [uploading](#sending-files) files.
 
 Files are stored separately from messages.
 A file on the Telegram servers is identified by a `file_id`, which is just a long string of characters.
+For example, it could look like `AgADBAADZRAxGyhM3FKSE4qKa-RODckQHxsoABDHe0BDC1GzpGACAAEC`.
 
-`AgADBAADZRAxGyhM3FKSE4qKa-RODckQHxsoABDHe0BDC1GzpGACAAEC` is an example of a `file_id`.
+### Identifiers for Receiving Files
+
+> Bots only receive file identifiers.
+> If they want to obtain file contents, they have to request them explicitly.
 
 Whenever your bot **receives** a message with a file, it will in fact not directly receive the complete file data, but only the `file_id` instead.
 If your bot actually wants to download the file, then it can do so by calling the `getFile` method ([Telegram Bot API reference](https://core.telegram.org/bots/api#getfile)).
 This method enables you to download the file by constructing a special, temporary, URL.
 Note that this URL is only guaranteed to be valid for 60 minutes, after which it may expire. In this case, you can simply call `getFile` again.
 
+Files can be received like [this](#receiving-files).
+
+### Identifiers for Sending Files
+
+> Sending files gives you a file identifier, too.
+
 Whenever your bot **sends** a message with a file, it will receive information about the sent message, including the `file_id` of the sent file.
 This means that all files the bot sees, both via sending or receiving, will make a `file_id` available to the bot.
+If you want to work with a file after your bot sees it, you should always store its `file_id`.
+
+> Use file identifiers whenever you can.
+> They are very efficient.
 
 When a bot sends a message, it can **specify a `file_id` that it has seen before**.
 This will allow it to send the identified file, without needing to upload the data for it.
-(To see how to upload your own files, [scroll down](#sending-files).)
+
 You can reuse the same `file_id` as often as you want, so you could send the same file to five different chats, using the same `file_id`.
 However, you must make sure to use the correct method---for example, you cannot use a `file_id` that identifies a photo when calling [`sendVideo`](https://core.telegram.org/bots/api#sendvideo).
 
+Files can be sent like [this](#sending-files).
+
+### Identifiers May Surprise You
+
+> File identifiers **only work for your bot**.
+> If another bot uses your file identifiers, it may randomly work and randomly crash and randomly kill innocent kittens.
```

Comment:
> it may randomly work

Didn’t know this! But yes, it’s technically possible.

Reply:
Indeed, and it happens that people run into this and then they are like “but it worked yesterday and now they're all dead” :( poor kittens, but we all know who's at fault here.